static/js/settings_users.js

```diff
@@ -682,10 +690,16 @@ function handle_bot_form($tbody, $status_field) {
             }
 
             settings_ui.do_settings_change(channel.patch, url, data, $status_field);
+            settings_ui.do_settings_change(
+                channel.patch,
+                url_for_role_change,
+                role_data,
+                $status_field,
+            );
```

Comment:
As a request for future PRs, please call attention when you're doing something quite weird like this (i.e. making write requests to two endpoints) -- it shouldn't be the reviewer's responsibility to notice this.  A code comment would be helpful, but also a comment on the PR is great.

Generally things like this that are effectively workarounds tend to be things we want to find a better solution to before merging a change.